pages/writing.md

```diff
@@ -45,23 +45,131 @@ juliaup status
 
 ## Development environments
 
+Depending on your experience and how you use Julia, some ways of writing code may be more familiar or useful to you than others.
+Data scientists commonly prefer notebook environments such as those provided by [Jupyter](https://jupyter.org/) through the [IJulia.jl](https://github.com/JuliaLang/IJulia.jl) package, or Julia's own reactive notebooks from [Pluto.jl](https://plutojl.org/).
+On the other hand, package developers or those who write a lot of scripts will prefer the more traditional programming environment provided by a text editor or integrated development environment (IDE).
+
+For some, the Julia REPL (Read Evaluate Print Loop) itself is all that's necessary.
+The REPL's primary function is to run code in "Julian" mode, but it also has a number of other modes that expand what can be done from inside Julia.
+Each mode is enterred by typing a character into the REPL from Julian mode, and can be exited by deleting this character with backspace.
+
+### Help mode (`?`)
+By entering a `?`, you can query information and metadata about Julia objects and unicode symbols simply by typing their name into the command line.
+For functions, types, and variables, the query fetches things such as documentation, type fields and supertypes, and in which file the object is defined.
+<!-- How do you do syntax highlighting for the Julia REPL? -->
+```
+help?> Int
+search: Int Int8 Int64 Int32 Int16 Int128 Integer intersect intersect!
+
+  Int64 <: Signed
+
+
+  64-bit signed integer type.
+```
+
+For unicode symbols, the query will return how to type the symbol in the REPL, which is useful when you copy-paste a symbol in without knowing its name, and fetch information about the object the symbol is bound to, just as above.
+
+### Pkg mode (`]`)
+Pkg mode is for managing environments and packages that is based on the Rust package manager "cargo".
+By pressing `]` in Julian mode, you can `add`, `update` (or `up`) and `remove` (or `rm`) packages, `activate` different local or global environments, and get the `status` (or `st`) of your current environment.
+
+More detail on using Pkg.jl and Pkg mode, see the [#Package-Management] section or the [#Setup] section of the [Sharing Julia code](./sharing.md#setup) post.
+
+<!-- Check whether the second link is needed after writing both sections. -->
+
+### Shell mode(`;`)
+Shell mode: Functions as a terminal inside Julia, you can also execude shell commands from Julian mode.
+Here's an example
+```julia
+an_example()
```

Comment:
Wtf ^^?

Reply:
ran out of energy

Reply:
What are the terminal niceties we might be interested in for this part? The one that comes to mind is git

Reply:
Git is one, but what I meant to show in this example is calling bash commands from inside a non-interactive Julia session. Calling shell commands from inside Julia means anything Julia can't do but bash can, can be done without needing to worry about interop.